discovery/bootstrapper.go

```diff
@@ -379,135 +379,135 @@ func (d *DNSSeedBootstrapper) SampleNodeAddrs(numAddrs uint32,
+					"back dns ssed (%v): %v", soaShim, err)
+				continue
 			}
 
-			log.Tracef("Retrieved SRV records from dns seed: %v",
-				spew.Sdump(addrs))
+			log.Tracef("Successfully queried fallback DNS seed")
+		}
 
-			// Next, we'll need to issue an A record request for
-			// each of the nodes, skipping it if nothing comes
-			// back.
-			for _, nodeSrv := range addrs {
-				if uint32(len(netAddrs)) >= numAddrs {
-					break search
-				}
+		log.Tracef("Retrieved SRV records from dns seed: %v",
+			spew.Sdump(addrs))
 
-				// With the SRV target obtained, we'll now
-				// perform another query to obtain the IP
-				// address for the matching bech32 encoded node
-				// key. We use the lndLookup function for this
-				// task.
-				bechNodeHost := nodeSrv.Target
-				addrs, err := d.net.LookupHost(bechNodeHost)
-				if err != nil {
-					return nil, err
-				}
-
-				if len(addrs) == 0 {
-					log.Tracef("No addresses for %v, skipping",
-						bechNodeHost)
-					continue
-				}
+		// Next, we'll need to issue an A record request for each of
+		// the nodes, skipping it if nothing comes back.
+		for _, nodeSrv := range addrs {
+			if uint32(len(netAddrs)) >= numAddrs {
+				break search
+			}
 
-				log.Tracef("Attempting to convert: %v", bechNodeHost)
+			// With the SRV target obtained, we'll now perform
+			// another query to obtain the IP address for the
+			// matching bech32 encoded node key. We use the
+			// lndLookup function for this task.
+			bechNodeHost := nodeSrv.Target
+			addrs, err := d.net.LookupHost(bechNodeHost)
+			if err != nil {
+				return nil, err
+			}
 
-				// If we have a set of valid addresses, then
-				// we'll need to parse the public key from the
-				// original bech32 encoded string.
-				bechNode := strings.Split(bechNodeHost, ".")
-				_, nodeBytes5Bits, err := bech32.Decode(bechNode[0])
-				if err != nil {
-					return nil, err
-				}
+			if len(addrs) == 0 {
+				log.Tracef("No addresses for %v, skipping",
+					bechNodeHost)
+				continue
+			}
 
-				// Once we have the bech32 decoded pubkey,
-				// we'll need to convert the 5-bit word
-				// grouping into our regular 8-bit word
-				// grouping so we can convert it into a public
-				// key.
-				nodeBytes, err := bech32.ConvertBits(
-					nodeBytes5Bits, 5, 8, false,
-				)
-				if err != nil {
-					return nil, err
-				}
-				nodeKey, err := btcec.ParsePubKey(
-					nodeBytes, btcec.S256(),
-				)
-				if err != nil {
-					return nil, err
-				}
+			log.Tracef("Attempting to convert: %v", bechNodeHost)
 
-				// If we have an ignore list, and this node is
-				// in the ignore list, then we'll go to the
-				// next candidate.
-				if ignore != nil {
-					nID := autopilot.NewNodeID(nodeKey)
-					if _, ok := ignore[nID]; ok {
-						continue
-					}
-				}
+			// If we have a set of valid addresses, then we'll need
+			// to parse the public key from the original bech32
+			// encoded string.
+			bechNode := strings.Split(bechNodeHost, ".")
```

Comment:
assert length is greater than 0? o/w a bug in the host could take down the daemon

Reply:
Done, also added a check to ensure it has a `.` within it as well. Aside from this, we should do some fuzzing in general on our `bech32` implementation.